funcframework/framework.go

```diff
@@ -206,6 +212,64 @@ func wrapEventFunction(fn interface{}) (http.Handler, error) {
 	}), nil
 }
 
+func wrapTypedFunction(fn interface{}) (http.Handler, error) {
+	err := validateTypedFunction(fn)
+	if err != nil {
+		return nil, err
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := readHTTPRequestBody(r)
+		if err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("%v", err))
+			return
+		}
+		argVal := reflect.New(reflect.TypeOf(fn).In(0))
+
+		if err := json.Unmarshal(body, argVal.Interface()); err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("Error: %s, while converting input type data: %s", err.Error(), string(body)))
+			return
+		}
+
+		defer recoverPanic(w, "user function execution")
+		funcReturn := reflect.ValueOf(fn).Call([]reflect.Value{
+			argVal.Elem(),
+		})
+
+		if len(funcReturn) >= 1 {
+			var errorVar error
```

Comment:
nit: What do you think of moving this logic to another function which returns (interface{}, error).  I am thinking of something like
```
func extractFnReturnValAndError(funcReturn) (returnVal, error)
```
Then you could do
```
returnVal, err := extractFnReturnValAndError(funcReturn)
if returnVal != nil {
    fmt.Fprintf(....)
}
if err != nil {
     writeHttpErrorResponse...
}
```
Let me know what you think.

Reply:
Good suggestion!